protoc-gen-gogo/generator/generator.go

```diff
@@ -1428,6 +1428,10 @@ func (g *Generator) generateEnum(enum *EnumDescriptor) {
 		g.PrintComments(fmt.Sprintf("%s,%d,%d", enum.path, enumValuePath, i))
 
 		name := ccPrefix + *e.Name
+		if gogoproto.IsEnumCustomName(e) {
```

Comment:
@awalterschulze Looks like `goproto_enum_prefix` is enabled by default. It wouldn't make much sense to emit a warning or error if `enum_customname` is also enabled since this would be triggered with common usage. I can still add it if you insist.